src/main/java/cms/gov/madie/measure/resources/TestCaseController.java

```diff
@@ -307,4 +304,50 @@ public ResponseEntity<List<String>> shiftMultiQiCoreTestCaseDates(
             .toList();
     return ResponseEntity.ok(failedTestCases);
   }
+
+  @PutMapping(ControllerUtil.TEST_CASES + "/copy-to")
+  public ResponseEntity<List<String>> copyToTestCaseDates(
+      @PathVariable String measureId,
+      @RequestParam(name = "targetMeasureId") String targetMeasureId,
+      @RequestBody List<String> testCaseIds,
+      Principal principal,
+      @RequestHeader("Authorization") String accessToken) {
+
+    Measure targetMeasure = measureService.findMeasureById(targetMeasureId);
+    Measure sourceMeasure = measureService.findMeasureById(measureId);
+    measureService.verifyAuthorization(principal.getName(), targetMeasure);
+    if (CollectionUtils.isEmpty(testCaseIds)) {
+      throw new InvalidRequestException("Test Case List cannot be empty");
+    }
+
+    if (!sameModelFamily(targetMeasure, sourceMeasure)) {
+      throw new InvalidRequestException("Target Measure has different model.");
+    }
+
+    List<TestCase> sourceTestCases =
+        testCaseService.findTestCasesByMeasureId(measureId).stream()
```

Comment:
can we use `sourceMeasure.getTestCases().stream...` instead since the data is already available?

Reply:
Yep. They were added at different times for different reasons. Didn't notice the duplication. Thanks!